bin/addmission.py

```diff
@@ -0,0 +1,419 @@
+		if data['m_iHTaskID'] > maxTaskId: 
+			maxTaskId = data['m_iHTaskID']
+	maxTaskId = maxTaskId + 1
+
+	lastNonInstanceTask = maxTaskId
+
+	def addMissionData(mission, taskInfo):
+		for k,v in missionData[0].items(): 
+			mission[k] = v
+			if isinstance(v, list):
+				mission[k] = [0] * len(v)
+
+		mission['m_iHMissionName'] = nameId
+		mission['m_iHJournalNPCID'] = taskInfo['journalNPC']
+		mission['m_iHMissionID'] = missionId
+
+		mission['m_iHMissionType'] = MISSION_TYPE
+		mission['m_iHDifficultyType'] = MISSION_DIFFICULTY
+		mission['m_iCTRReqLvMin'] = REQUIRED_LEVEL
+
+		mission['m_iHTaskType'] = taskInfo['taskType']
+		mission['m_iHCurrentObjective'] = createMissionString(taskInfo['objective'])
+
+		if 'initialMessage' in taskInfo:
+			mission['m_iSTMessageType'] = 2
+			mission['m_iSTMessageTextID'] = createMissionString(taskInfo['initialMessage']['text'])
+			mission['m_iSTMessageSendNPC'] = taskInfo['initialMessage']['npc']
+
+		if 'initialDialog' in taskInfo:
+			mission['m_iSTDialogBubble'] = createMissionString(taskInfo['initialDialog']['text'])
+			mission['m_iSTDialogBubbleNPCID'] = taskInfo['initialDialog']['npc']
+		
+		if 'successMessage' in taskInfo:
+			mission['m_iSUMessageType'] = 2
+			mission['m_iSUMessagetextID'] = createMissionString(taskInfo['successMessage']['text']) # lowercase t 🙄
+			mission['m_iSUMessageSendNPC'] = taskInfo['successMessage']['npc']
+
+		if 'successDialog' in taskInfo:
+			mission['m_iSUDialogBubble'] = createMissionString(taskInfo['successDialog']['text'])
+			mission['m_iSUDialogBubbleNPCID'] = taskInfo['successDialog']['npc']
+			
+		if 'startItems' in taskInfo:
+			for i in range(min(3, len(taskInfo['startItems']['items']))):
+				mission['m_iSTItemID'][i] = taskInfo['startItems']['items'][i]
+				mission['m_iSTItemNumNeeded'][i] = taskInfo['startItems']['count'][i]
+				if taskInfo['taskType'] != 5:
+					# If there are no mob drops just throw these values instead just for fun (Putting them in CSUItems makes the counter show up on the UI)
+					mission['m_iCSUItemID'][i] = taskInfo['startItems']['items'][i]
+					mission['m_iCSUItemNumNeeded'][i] = taskInfo['startItems']['count'][i]
+
+		if 'successItems' in taskInfo:
+			for i in range(min(3, len(taskInfo['successItems']['items']))):
+				mission['m_iSUItem'][i] = taskInfo['successItems']['items'][i]
+				mission['m_iSUInstancename'][i] = taskInfo['successItems']['count'][i]
+
+		if 'failItems' in taskInfo:
+			for i in range(min(3, len(taskInfo['failItems']['items']))):
+				mission['m_iFItemID'][i] = taskInfo['failItems']['items'][i]
+				mission['m_iFItemNumNeeded'][i] = taskInfo['failItems']['count'][i]
+
+		for i in range(min(2, len(MISSION_PREREQS))):
+			mission['m_iCSTReqMission'][i] = MISSION_PREREQS[i]
+
+		journalId = createJournalEntry(taskInfo['taskDesc'])
+		mission['m_iSTJournalIDAdd'] = journalId
+		mission['m_iSUJournalIDAdd'] = journalId # NOTE in some missions these are different? but it's a pretty useless thing

```

Comment:
```suggestion
		mission['m_iSUJournaliDAdd'] = journalId # NOTE in some missions these are different? but it's a pretty useless thing

```
it's actually like that in tabledata /sigh